pkg/policy/repository.go

```diff
@@ -460,6 +462,8 @@ func (r ruleSlice) UpdateRulesEndpointsCaches(endpointsToBumpRevision *EndpointS
 	}
 
 	endpointsToBumpRevision.ForEach(policySelectionWG, func(epp Endpoint) {
+		epp.UnconditionalRLock()
```

Comment:
Should this be a lockalive, and no-op on error? I'm not sure what the side-effects are to applying `updateEndpointsCaches` on disconnected endpoints.

Reply:
I think you're right.

Reply:
seems like this caused endpoints to deadlock, trying to find out why